[Bug]: TRTLLM Attention Failure with DP/EP

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>


### 🐛 Describe the bug

Deploy with:

```
MODEL := "nvidia/Qwen3-30B-A3B-NVFP4"

GPUS := "2"
PORT := "8001"

launch_dp_ep_:
	VLLM_USE_FLASHINFER_MOE_FP4=1 VLLM_FLASHINFER_MOE_BACKEND=masked_gemm chg run --gpus {{GPUS}} -- vllm serve {{MODEL}} -dp {{GPUS}} --enable-expert-parallel --port {{PORT}} --all2all-backend deepep_low_latency

eval:
	lm_eval \
		--model local-completions \
		--tasks gsm8k \
		--model_args "model={{MODEL}},base_url=http://localhost:{{PORT}}/v1/completions,num_concurrent=1000,tokenized_requests=False" --limit 100
```

you will hit
```
(EngineCore_DP1 pid=2893355)   File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
(EngineCore_DP1 pid=2893355)     self.run()
(EngineCore_DP1 pid=2893355)   File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
(EngineCore_DP1 pid=2893355)     self._target(*self._args, **self._kwargs)
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/engine/core.py", line 939, in run_engine_core
(EngineCore_DP1 pid=2893355)     raise e
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/engine/core.py", line 928, in run_engine_core
(EngineCore_DP1 pid=2893355)     engine_core.run_busy_loop()
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/engine/core.py", line 1368, in run_busy_loop
(EngineCore_DP1 pid=2893355)     self.execute_dummy_batch()
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/engine/core.py", line 578, in execute_dummy_batch
(EngineCore_DP1 pid=2893355)     self.model_executor.execute_dummy_batch()
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/executor/abstract.py", line 229, in execute_dummy_batch
(EngineCore_DP1 pid=2893355)     self.collective_rpc("execute_dummy_batch")
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/executor/uniproc_executor.py", line 75, in collective_rpc
(EngineCore_DP1 pid=2893355)     result = run_method(self.driver_worker, method, args, kwargs)
(EngineCore_DP1 pid=2893355)              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/serial_utils.py", line 461, in run_method
(EngineCore_DP1 pid=2893355)     return func(*args, **kwargs)
(EngineCore_DP1 pid=2893355)            ^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/worker/gpu_worker.py", line 670, in execute_dummy_batch
(EngineCore_DP1 pid=2893355)     self.model_runner._dummy_run(1, uniform_decode=True)
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/.venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 120, in decorate_context
(EngineCore_DP1 pid=2893355)     return func(*args, **kwargs)
(EngineCore_DP1 pid=2893355)            ^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/worker/gpu_model_runner.py", line 4389, in _dummy_run
(EngineCore_DP1 pid=2893355)     attn_metadata, _ = self._build_attention_metadata(
(EngineCore_DP1 pid=2893355)                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/worker/gpu_model_runner.py", line 1775, in _build_attention_metadata
(EngineCore_DP1 pid=2893355)     _build_attn_group_metadata(kv_cache_gid, attn_gid, cm)
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/worker/gpu_model_runner.py", line 1726, in _build_attn_group_metadata
(EngineCore_DP1 pid=2893355)     attn_metadata_i = builder.build(
(EngineCore_DP1 pid=2893355)                       ^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355)   File "/home/robertgshaw2-redhat/vllm/vllm/v1/attention/backends/flashinfer.py", line 1062, in build
(EngineCore_DP1 pid=2893355)     assert num_decode_tokens % num_decodes == 0, (
(EngineCore_DP1 pid=2893355)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(EngineCore_DP1 pid=2893355) AssertionError: TRTLLM decode requires uniform query lengths per request.
```


### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/vllm/v1/worker/gpu_model_runner.py
+++ b/vllm/v1/worker/gpu_model_runner.py
@@ -4787,9 +4787,7 @@ class GPUModelRunner(
             pad_attn = cudagraph_runtime_mode == CUDAGraphMode.FULL
             attn_metadata, _ = self._build_attention_metadata(
                 num_tokens=num_tokens_unpadded,
-                num_tokens_padded=num_tokens_padded
-                if pad_attn
-                else num_tokens_unpadded,
+                num_tokens_padded=num_tokens_padded if pad_attn else None,
                 num_reqs=num_reqs_padded,
                 max_query_len=max_query_len,
                 ubatch_slices=ubatch_slices_padded if pad_attn else ubatch_slices,

--- a/vllm/v1/worker/gpu_model_runner.py
+++ b/vllm/v1/worker/gpu_model_runner.py
@@ -4787,6 +4787,9 @@ class GPUModelRunner(
             pad_attn = cudagraph_runtime_mode == CUDAGraphMode.FULL
             attn_metadata, _ = self._build_attention_metadata(
                 num_tokens=num_tokens_unpadded,
+                num_tokens_padded=num_tokens_padded
+                if pad_attn
+                else num_tokens_unpadded,
                 num_reqs=num_reqs_padded,
                 max_query_len=max_query_len,
                 ubatch_slices=ubatch_slices_padded if pad_attn else ubatch_slices,

